packages/pigeon/platform_tests/ios_unit_tests/ios/RunnerTests/MultipleArityTest.m

```diff
@@ -0,0 +1,36 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/Flutter.h>
+#import <XCTest/XCTest.h>
+#import "HandlerBinaryMessenger.h"
+#import "multiple_arity.gen.h"
+
+///////////////////////////////////////////////////////////////////////////////////////////
+@interface MultipleAritytest : XCTestCase
+@end
+
+///////////////////////////////////////////////////////////////////////////////////////////
+@implementation MultipleAritytest
+
+- (void)testSimple {
+  HandlerBinaryMessenger *binaryMessenger =
+      [[HandlerBinaryMessenger alloc] initWithCodec:MultipleArityHostApiGetCodec()
+                                            handler:^id _Nullable(NSArray *_Nonnull args) {
+                                              return @([args[0] intValue] - [args[1] intValue]);
+                                            }];
+  MultipleArityFlutterApi *api =
+      [[MultipleArityFlutterApi alloc] initWithBinaryMessenger:binaryMessenger];
+  XCTestExpectation *expectation = [self expectationWithDescription:@"subtraction"];
+  [api subtract:@(30)
+               y:@(10)
```

Comment:
I know it's just a test, but maybe the example multiple-arity function could be something that's not intrinsically a poorly-named function? The ObjC generated really highlights it, but even in other languages `subtract(int x, int y)` is a terrible function because there's nothing anywhere in the declaration that indicates the—critically important—order. Just something that's not order-dependent (e.g., `multiply`) would be a big improvement.

Reply:
I purposefully chose a function that wasn't transitive to make sure that the order was being asserted.  I originally wrote this as `add` but it didn't test the code as well as `subtract`.

Reply:
Maybe it should be something whose arguments are different types? That would not only test order, but would make order problems a compile error rather than a runtime error. E.g., `getDisplayString(int value, String unit)` (`=> '$value$unit'`)

Reply:
Runtime versus compilation error isn't a problem for unit tests since a runtime error happens a fraction of a second later.  It's not trivial to change this example, 30-40 minutes.  There are precedents for functions like this, `atan2` and `pow`.  Having a poor objc output will be addressed in the custom selector PR.

Reply:
To clarify: my first comment is the thing I think is the problem here (the terrible function name). The second was a suggestion for a specific fix that in addition to resolving that, would both address your response (that the function needs to assert order) and have secondary benefits (compile-time error).

> There are precedents for functions like this, atan2 and pow

`atan2` does indeed have this problem. `pow` does not because `num pow(num x, num exponent)` has unambiguous argument names—nobody could read that and not know which of the two arguments is the exponent. I wouldn't have an issue with this example if it were `pow(int x, int exponent)`.

But if you don't care that the example API in your tests is a naming anti-pattern, okay.